src/mpi/init/mpir_finalize.c

```diff
@@ -33,8 +33,7 @@ void MPIR_Add_finalize(int (*f) (void *), void *extra_data, int priority)
          * MPIR_Process.mpich_state to decide how to signal the error */
         (void) MPL_internal_error_printf("overflow in finalize stack! "
                                          "Is MAX_FINALIZE_FUNC too small?\n");
-        if (MPL_atomic_load_int(&MPIR_Process.mpich_state) != MPICH_MPI_STATE__PRE_INIT &&
-            MPL_atomic_load_int(&MPIR_Process.mpich_state) != MPICH_MPI_STATE__POST_FINALIZED) {
+        if (MPIR_Err_initialized()) {
             MPID_Abort(NULL, MPI_SUCCESS, 13, NULL);
```

Comment:
You are looking at the code rather than the usage. The code checks for state, but the usage is for whether error utility is ready. I believe we should name the function according to its usage as it makes sense to the user. The code can change based on internal design. Here the connection between `MPICH_MPI_STATE__PRE_INIT` or `MPICH_MPI_STATE__POST_FINALIZED` and whether the error utility can be used is not obvious and in fact I would very much refactor it away. Adding a specific state variable such as `MPIR_err_is_initialized` is one such refactor.

Reply:
I am marking this as resolved to clear my todo list. Feel free to unresolved them if needed.

Reply:
Since you mentioned `MPIR_Err_uninitialized` in a later comment, I agree `MPIR_Err_initialized` is confusing. I have renamed it to `MPIR_Errutil_is_initialized` now, hopefully, this is clearer.